### PR TITLE
Docs: improve/cleanup reference from/to recwarn

### PR DESCRIPTION
**Pull request by soxofaan**
- [X] Allow maintainers to push and squash when merging my commits. Please uncheck this if you prefer to squash the commits yourself.
- [x] Create a new changelog file in the `changelog` folder, with a name like `<ISSUE NUMBER>.<TYPE>.rst`. See [changelog/README.rst](https://github.com/pytest-dev/pytest/blob/main/changelog/README.rst) for details.
- [x] Add yourself to `AUTHORS` in alphabetical order.

### CODE REVIEW

**Comment by soxofaan** [comment]
I was looking up some docs around the `recwarn` fixture today and had a hard time getting to the point. This PR tries to fix some problems in the docs around that:

- replace some raw http links with sphinx references
- when mentioning `recwarn`, point to the API docs instead of a section that barely explains something about `recwarn`
- from the `recwarn` API docs, provide cross-reference to a how-to section about `recwarn` instead of something loosely related
- `recwarn` API docs: also document `__getitem__`, `__iter__` and `__len__` (which are shown in the how to, but not explained)

File: changelog/12866.doc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Improved cross-references concerning the :func:`recwarn <_pytest.recwarn.recwarn>` fixture.
```
Comment (nicoddemus) [inline changelog/12866.doc.rst:1]:
```suggestion
Improved cross-references concerning the :fixture:`recwarn` fixture.
```

File: doc/en/how-to/capture-warnings.rst [hunk 2]
```diff
@@ -195,7 +195,7 @@ user code and third-party libraries, as recommended by :pep:`565`.
 This helps users keep their code modern and avoid breakages when deprecated warnings are effectively removed.
 
 However, in the specific case where users capture any type of warnings in their test, either with
-:func:`pytest.warns`, :func:`pytest.deprecated_call` or using the :ref:`recwarn <recwarn>` fixture,
+:func:`pytest.warns`, :func:`pytest.deprecated_call` or using the :func:`recwarn <_pytest.recwarn.recwarn>` fixture,
```
Comment (nicoddemus) [inline doc/en/how-to/capture-warnings.rst:198]:
```suggestion
:func:`pytest.warns`, :func:`pytest.deprecated_call` or using the :fixture:`recwarn` fixture,
```

File: doc/en/how-to/capture-warnings.rst [hunk 3]
```diff
@@ -332,10 +332,10 @@ additional information:
     assert record[0].message.args[0] == "another warning"
 
 Alternatively, you can examine raised warnings in detail using the
-:ref:`recwarn <recwarn>` fixture (see below).
+:func:`recwarn <_pytest.recwarn.recwarn>` fixture (see :ref:`below <recwarn>`).
```
Comment (nicoddemus) [inline doc/en/how-to/capture-warnings.rst:335]:
```suggestion
:fixture:`recwarn` fixture (see :ref:`below <recwarn>`).
```

File: doc/en/how-to/capture-warnings.rst [hunk 4]
```diff
@@ -332,10 +332,10 @@ additional information:
...
 
 
-The :ref:`recwarn <recwarn>` fixture automatically ensures to reset the warnings
+The :func:`recwarn <_pytest.recwarn.recwarn>` fixture automatically ensures to reset the warnings
```
Comment (nicoddemus) [inline doc/en/how-to/capture-warnings.rst:338]:
```suggestion
The :fixture:`recwarn` fixture automatically ensures to reset the warnings
```

File: doc/en/how-to/capture-warnings.rst [hunk 5]
```diff
@@ -345,8 +345,8 @@ filter at the end of the test, so no global state is leaked.
 Recording warnings
 ------------------
 
-You can record raised warnings either using :func:`pytest.warns` or with
-the ``recwarn`` fixture.
+You can record raised warnings either using the :func:`pytest.warns` context manager or with
+the :func:`recwarn <_pytest.recwarn.recwarn>` fixture.
```
Comment (nicoddemus) [inline doc/en/how-to/capture-warnings.rst:349]:
```suggestion
the :fixture:`recwarn` fixture.
```

File: doc/en/how-to/capture-warnings.rst [hunk 6]
```diff
@@ -361,7 +361,7 @@ pass no arguments as the expected warning type and it will default to a generic
     assert str(record[0].message) == "user"
     assert str(record[1].message) == "runtime"
 
-The ``recwarn`` fixture will record warnings for the whole function:
+The :func:`recwarn <_pytest.recwarn.recwarn>` fixture will record warnings for the whole function:
```
Comment (nicoddemus) [inline doc/en/how-to/capture-warnings.rst:364]:
```suggestion
The :fixture:`recwarn` fixture will record warnings for the whole function:
```

File: doc/en/how-to/capture-warnings.rst [hunk 7]
```diff
@@ -377,12 +377,11 @@ The ``recwarn`` fixture will record warnings for the whole function:
         assert w.filename
         assert w.lineno
 
-Both ``recwarn`` and :func:`pytest.warns` return the same interface for recorded
-warnings: a WarningsRecorder instance. To view the recorded warnings, you can
+Both the :func:`recwarn <_pytest.recwarn.recwarn>` fixture and the :func:`pytest.warns` context manager return the same interface for recorded
```
Comment (nicoddemus) [inline doc/en/how-to/capture-warnings.rst:380]:
```suggestion
Both the :fixture:`recwarn` fixture and the :func:`pytest.warns` context manager return the same interface for recorded
```

**Review by nicoddemus — CHANGES_REQUESTED** [review]

File: changelog/12866.doc.rst (hunk 1 above)
Comment (soxofaan) [reply changelog/12866.doc.rst]: aha great, so there is a sphinx role for fixtures, nice

**Review by nicoddemus — APPROVED** [review]
Thanks!

**Pull request merged**